modules/dnn/src/layers/scale_layer.cpp

```diff
@@ -78,64 +78,95 @@ class ScaleLayerImpl CV_FINAL : public ScaleLayer
         Mat &outBlob = outputs[0];
         // There is a mode when we multiply a first blob by a second one
         // instead of trainable weights.
-        Mat weights = hasWeights ? (blobs.empty() ? inputs[1] : blobs[0]).reshape(1, 1) : Mat();;
+        Mat weights = hasWeights ? (blobs.empty() ? inputs[1] : blobs[0]).reshape(1, 1) : Mat();
         Mat bias = hasBias ? (blobs.empty() ? inputs[1] : blobs.back()).reshape(1, 1) : Mat();
 
         MatShape inpShape = shape(inpBlob);
+        const int numSamples = inpShape[0];
+        const int numWeightsSamples = (hasWeights || hasBias) && blobs.empty() ? numSamples : 1;
         const int numWeights = !weights.empty() ? weights.total() : bias.total();
+        const int numWeightsPerSample = numWeights / numWeightsSamples;
         CV_Assert(numWeights != 0);
         if (hasWeights && hasBias)
             CV_CheckEQ(weights.total(), bias.total(), "Incompatible weights/bias blobs");
 
-        int endAxis;
+        axis = clamp(axis , inpBlob.dims) ? clamp(axis , inpBlob.dims) : 1;
+        CV_Assert(axis > 0 && axis < inpBlob.dims);
+
+        int endAxis = 1;
         for (endAxis = axis + 1; endAxis <= inpBlob.dims; ++endAxis)
         {
-            if (total(inpShape, axis, endAxis) == numWeights)
+            if (total(inpShape, axis, endAxis) == numWeightsPerSample)
                 break;
         }
-        CV_Assert(total(inpShape, axis, endAxis) == numWeights);
+        endAxis = (numWeightsPerSample == 1) ? 1 : endAxis;
+
+        CV_Assert(total(inpShape, axis, endAxis) == numWeightsPerSample);
         CV_Assert(!hasBias || numWeights == bias.total());
```

Comment:
Checking broadcasting options. if size of weights is either N or 1, then endAxis=1. 
ex - NCHW ( input dims ), N/1 ( weights ). then endAxis = 1.
otherwise calculate endAxis from condition. 
`if (total(inpShape, axis, endAxis) == numWeightsPerSample)`